File: frontend/src/types/BoardType.ts
export enum BoardType {
  SMALL = 'SMALL',
  STANDARD = 'STANDARD', 
  LARGE = 'LARGE',
  EXTRA_LARGE = 'EXTRA_LARGE',
  HUGE = 'HUGE',
  MASSIVE = 'MASSIVE'
}

export interface BoardTypeInfo {
  type: BoardType;
  size: number;
  description: string;
}

export const BOARD_TYPES: BoardTypeInfo[] = [
  { type: BoardType.SMALL, size: 10, description: '10x10 - Nhỏ' },
  { type: BoardType.STANDARD, size: 15, description: '15x15 - Tiêu chuẩn' },
  { type: BoardType.LARGE, size: 20, description: '20x20 - Lớn' },
  { type: BoardType.EXTRA_LARGE, size: 25, description: '25x25 - Rất lớn' },
  { type: BoardType.HUGE, size: 30, description: '30x30 - Khổng lồ' },
  { type: BoardType.MASSIVE, size: 40, description: '40x40 - Siêu lớn' }
];

File: backend/src/main/java/com/example/carogame/model/BoardType.java
package com.example.carogame.model;

public enum BoardType {
    SMALL(10, "10x10 - Nhỏ"),
    STANDARD(15, "15x15 - Tiêu chuẩn"),
    LARGE(20, "20x20 - Lớn"),
    EXTRA_LARGE(25, "25x25 - Rất lớn"),
    HUGE(30, "30x30 - Khổng lồ"),
    MASSIVE(40, "40x40 - Siêu lớn");

    private final int size;
    private final String description;

    BoardType(int size, String description) {
        this.size = size;
        this.description = description;
    }

    public int getSize() {
        return size;
    }

    public String getDescription() {
        return description;
    }

    public static BoardType fromSize(int size) {
        for (BoardType type : values()) {
            if (type.size == size) {
                return type;
            }
        }
        return STANDARD; // Default fallback
    }
}
